Home page

pettinarip opened the pull request:
## Description

Implements the home page.

pettinarip committed:
initial home page migration

pettinarip committed:
refactor community events data fetch to be done on build time

pettinarip committed:
adapt hero image

pettinarip committed:
fix images

pettinarip committed:
remove files from tsconfig

pettinarip committed:
implement cache on fetch call

pettinarip committed:
throw warning instead of an error if in dev mode

pettinarip committed:
add lastDeployDate to page props

wackerow commented:
Odd regression with smart contract syntax highlighing... I'm seeing the following when clicking the 1st, 2nd or 4th contract example the _first_ time... closing modal and re-opening returns appropriate highlighting:

PR:
<img width="1643" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/a3fd276e-7e89-4074-b586-b3c8aa5e56d8">

Canonical (https://ethereum.org):
<img width="1149" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/d748d0cf-dade-4f83-bbdf-a6cd2875c928">

wackerow commented:
Small differences noted here... not sure why a few of the events and dates and different between these; may be a non-issue... Also noting our Discord link has lost the external arrow. This was previously obtained using some manual logic since we're linking to `/discord/` which isn't normally interpreted as external.

Canonical (https://ethereum.org):
<img width="1494" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/8ea73c66-d6d3-4ded-b288-978d9f99ed24">

PR:
<img width="1492" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/b68dc825-8564-4dca-873a-9a561b520da8">

---

Image sizes are also a little different in explore section:

Canonical (https://ethereum.org):
<img width="1456" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/04d726c3-bebb-4614-b1d1-9756e1429427">

PR:
<img width="1477" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/106e7509-1699-4335-9f2c-e2c3cd96c5e0">

pettinarip commented:
Thanks for the review @wackerow. Will look into those issues and let you know.

pettinarip committed:
Merge branch 'dev' into page-home

pettinarip committed:
install deps after update

pettinarip committed:
fixes after merge

pettinarip committed:
import prism-solidity outside of the useeffect

pettinarip committed:
lint fix

pettinarip committed:
treat discord invite href as an external link

pettinarip committed:
fix image size in the ActionCard component

pettinarip commented:
@wackerow fixed the syntax highlighing issue. In short, we were initializing the prism-solidity module in a useEffect, so it was not loaded at the first render. I've put it outside of the component to avoid this problem.

> Small differences noted here... not sure why a few of the events and dates and different between these; may be a non-issue... Also noting our Discord link has lost the external arrow. This was previously obtained using some manual logic since we're linking to `/discord/` which isn't normally interpreted as external.

The difference in events and dates were because this build was triggered a few days ago and the calendar was updated in the middle.

I've also added the discord invite link to be considered as an external link.

> Image sizes are also a little different in explore section:

Fixed.

pettinarip committed:
Merge branch 'dev' into page-home

pettinarip committed:
fixes after update

pettinarip committed:
rename function and add documentation

pettinarip committed:
revalidate home page every day

corwintines committed:
Merge branch 'dev' into page-home

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -96,13 +100,13 @@ const CommunityEvents = () => {
         <Box pe={8} ps={{ base: 8, lg: 0 }}>
           <OldHeading>
-            <Translation id="community-events-content-heading" />
+            <Translation id="page-index:community-events-content-heading" />
```
can use hook directly

```suggestion
{t("page-index:community-events-content-heading")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -96,13 +100,13 @@ const CommunityEvents = () => {
           </OldHeading>
           <Text>
-            <Translation id="community-events-content-1" />
+            <Translation id="page-index:community-events-content-1" />
```
```suggestion
{t("page-index:community-events-content-1")}
```

corwintines replied:
I don't think this can happen right now. The t() way of doing things doesn't render html elements in the strings properly, and the Translation component does.

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -96,13 +100,13 @@ const CommunityEvents = () => {
           </Text>
           <Text>
-            <Translation id="community-events-content-2" />
+            <Translation id="page-index:community-events-content-2" />
```
```suggestion
{t("page-index:community-events-content-2")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -118,54 +122,44 @@ const CommunityEvents = () => {
+              </Box>
+            ) : (
+              <Text fontSize="3xl" fontWeight="bold" mb={8}>
+                <Translation id="page-index:community-events-no-events-planned" />
```
```suggestion
{t("page-index:community-events-no-events-planned")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -174,54 +168,38 @@ const CommunityEvents = () => {
         >
           <Text fontSize="lg" fontWeight="bold" mb={2}>
-            <Translation id="community-events-upcoming-calls" />
+            <Translation id="page-index:community-events-upcoming-calls" />
```
```suggestion
{t("page-index:community-events-upcoming-calls")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -174,54 +168,38 @@ const CommunityEvents = () => {
           ) : (
             <Text mx="auto">
-              <Translation id="community-events-no-upcoming-calls" />
+              <Translation id="page-index:community-events-no-upcoming-calls" />
```
```suggestion
{t("page-index:community-events-no-upcoming-calls")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -174,54 +168,38 @@ const CommunityEvents = () => {
           )}
           <Text fontSize="lg" fontWeight="bold" mb={2} mt={4}>
-            <Translation id="community-events-previous-calls" />
+            <Translation id="page-index:community-events-previous-calls" />
```
```suggestion
{t("page-index:community-events-previous-calls")}
```

nhsz commented on src/components/CommunityEvents/index.tsx:
```diff
@@ -174,54 +168,38 @@ const CommunityEvents = () => {
           ) : (
             <Text mx="auto">
-              <Translation id="community-events-there-are-no-past-calls" />
+              <Translation id="page-index:community-events-there-are-no-past-calls" />
```
```suggestion
{t("page-index:community-events-there-are-no-past-calls")}
```

nhsz commented on src/components/Hero/HomeHero/index.tsx:
```diff
@@ -32,13 +36,13 @@ const HomeHero = ({ heroImgSrc }: HomeHeroProps) => {
           <VStack spacing="6">
             <Heading as="h1" size="2xl">
-              <Translation id="page-index-title" />
+              <Translation id="page-index:page-index-title" />
```
```suggestion
{t("page-index:page-index-title")}
```

nhsz commented on src/components/Hero/HomeHero/index.tsx:
```diff
@@ -32,13 +36,13 @@ const HomeHero = ({ heroImgSrc }: HomeHeroProps) => {
             </Heading>
             <Text size="xl">
-              <Translation id="page-index-description" />
+              <Translation id="page-index:page-index-description" />
```
```suggestion
{t("page-index:page-index-description")}
```

nhsz commented on src/components/Hero/HomeHero/index.tsx:
```diff
@@ -32,13 +36,13 @@ const HomeHero = ({ heroImgSrc }: HomeHeroProps) => {
             </Text>
             <ButtonLink to="/learn/">
-              <Translation id="page-index-title-button" />
+              <Translation id="page-index:page-index-title-button" />
```
```suggestion
{t("page-index:page-index-title-button")}
```

corwintines committed:
Update imports

nhsz commented on src/lib/api/calendarEvents.ts:
```diff
@@ -0,0 +1,80 @@
+  const calendarId = process.env.GOOGLE_CALENDAR_ID
+
+  try {
+    const futureEventsReq = await axios.get<{ items: ReqCommunityEvent[] }>(
```
we should use native `fetch` instead of axios, and remove this dep

nhsz commented on src/lib/utils/cache.ts:
file should be renamed as `runOnlyOnce.ts` too

nhsz commented on src/lib/utils/cache.ts:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Run an async function only once and cache the result
```
```suggestion
 * Run an async function only once and cache the result during build time to avoid multiple calls. The purpose is to avoid hitting external APIs rate limits.
```

nhsz commented on src/lib/interfaces.ts:
```diff
@@ -161,3 +161,26 @@ export interface MdPageContent {
+  date: string
+  title: string
+  calendarLink: string
+  pastEventLink: string | undefined
```
```suggestion
  pastEventLink?: string
```

nhsz commented on src/components/Hero/HomeHero/index.tsx:
```diff
@@ -2,25 +2,29 @@ import * as React from "react"
 
 const HomeHero = ({ heroImgSrc }: HomeHeroProps) => {
-  const { t } = useTranslation()
+  const { t } = useTranslation("common")
```
should be `"page-index"`here instead of `"common"`? and  `"page-index"` could be removed maybe

nhsz commented on src/pages/index.tsx:
```diff
@@ -0,0 +1,611 @@
+      communityEvents,
+      lastDeployDate,
+    },
+    revalidate: BASE_TIME_UNIT * 24,
```
```suggestion
    revalidate: BASE_TIME_UNIT * 24, // Rebuilds page daily
```

nhsz commented on src/pages/index.tsx:
```diff
@@ -0,0 +1,611 @@
+import HomeHero from "../components/Hero/HomeHero"
+import PageMetadata from "../components/PageMetadata"
+// TODO: migrate stats fetching on build time
+// import StatsBoxGrid from "../components/StatsBoxGrid"
```
seems this component is not being used here

corwintines replied:
I think this should stay in. @pettinarip mentioned this needs migration still in dev sync today

nhsz commented on src/pages/index.tsx:
```diff
@@ -0,0 +1,611 @@
+            <Translation id="page-index:page-index-network-stats-subtitle" />
+          </SectionDecription>
+        </ContentBox>
+        {/* // TODO: migrate stats fetching the necessary data on build time */}
```
```suggestion
```

corwintines replied:
I think this should stay in. @pettinarip mentioned this needs migration still in dev sync today

nhsz commented on src/pages/index.tsx:
```diff
@@ -0,0 +1,611 @@
+import CommunityEvents from "../components/CommunityEvents"
+import HomeHero from "../components/Hero/HomeHero"
+import PageMetadata from "../components/PageMetadata"
+// TODO: migrate stats fetching on build time
```
```suggestion
```

corwintines replied:
I think this should stay in. @pettinarip mentioned this needs migration still in dev sync today

nhsz reviewed (changes requested):
Awesome, good job, asking for a couple small adjustments

corwintines committed:
Change requests

corwintines committed:
Merge branch 'dev' into page-home

corwintines committed:
lint

corwintines reviewed (approved):
Applied changes from @nhsz. LGTM. Going to merge. Nice one @pettinarip!
